pylint/checkers/similar.py

```diff
@@ -557,7 +557,7 @@ def stripped_lines(
     ignore_docstrings: bool,
     ignore_imports: bool,
     ignore_signatures: bool,
-    line_enabled_callback: Optional[Callable] = None,
+    line_enabled_callback: Callable | None = None,
```

Comment:
I fought with this locally and finally just `no-verified` it. Why is this syntax okay with our minimum python version of 3.7.2? (Well, 3.6.2, since we are talking about backporting, potentially.)

Reply:
oh, did we future import annotations everywhere? ah. but still a problem with 3.6, then.

Reply:
I guess we don't have to backport this if only one user reported it and 2.14 is coming soon.

Reply:
```suggestion
    line_enabled_callback: Callable[[PyLinter, str, int | None], bool] | None = None,
```

I think? If we turn on ``--strict`` I think `mypy` will complain about unspecified generics like these.

Reply:
Cool. LMK if I should move the release note to 2.14. If we backport I guess we can just use different type annotation syntax.